### DISCUSSION TITLE
JSON/native specimen to test ALL pandoc elements

### DISCUSSION THREAD

**Discussion by massifrg** [jgm/pandoc#10188] [Q&A] [answered]
Is there a specimen document that contains every Meta, Block and Inline supported by Pandoc types, to be used as a testsuite?

It should be in native or JSON format, since they are the most complete.

**Comment by jgm** [comment, marked as answer]
`test/testsuite.native` should have *most* of the types represented.  Possibly not all, though.
(e.g. line blocks are tested in `test/markdown-reader-more.native`.

**Comment by massifrg** [comment]
I made a review of all the elements in `test/testsuite.native`, that I leave here for future reference.

The elements that are not in `testsuite.native` can be found in other `.native` files in the same `test/` directory.

Blocks:
- Plain
- Para
- ~~LineBlock~~ (present in `rst-reader.native`)
- CodeBlock
- RawBlock
- BlockQuote
- OrderedList
- BulletList
- DefinitionList
- Header
- HorizontalRule	
- ~~Table~~ (see e.g. `html-reader.native`)
- Figure
- Div

Inlines:
- Str
- Emph
- ~~Underline~~ (see e.g. `html-reader.native`)
- Strong
- Strikeout
- Superscript
- Subscript
- ~~SmallCaps~~ (see e.g. `html-reader.native`)
- Quoted
- ~~Cite~~ (in `markdown-citations.native`)
- Code
- Space
- SoftBreak	
- LineBreak	
- Math
- RawInline
- Link
- Image
- Note
- ~~Span~~ (see e.g. `html-reader.native`)

Meta:
- ~~MetaMap~~ (in `jats-reader.native`)
- MetaList
- ~~MetaBool~~ (no `test/*.native` file contains it)
- ~~MetaString~~ (see e.g. `jats-reader.native`)
- MetaInlines
- ~~MetaBlocks~~ (in `rst-reader.native`)

So the minimal collection of `.native` files that cover all the elements (but `MetaBool`) is:
- `testsuite.native`: most of the elements
- `html-reader.native`: Table, Underline, SmallCaps, Span
- `rst-reader.native`: LineBlock, MetaBlocks
- `jats-reader.native`: MetaMap, MetaString
- `markdown-citations.native`: Cite

**Reply by jgm** [reply]
It might be nice to have one compact file that has everything.
